header-validator/validate-json.js

```diff
@@ -269,6 +270,16 @@ const aggregationCoordinator = string((state, value) => {
   state.error(`must match '${awsCloud}' (case-sensitive)`)
 })
 
+const aggregatableDedupKeys = list(
+  (state, value) =>
+    state.validate(value, {
+      deduplication_key: optional(uint64),
```

Comment:
Is there a benefit to making this field optional? Wouldn't we just want to omit the list entry instead?

Reply:
It may be useful if one want to use two filters and only set the dedup key if the first filter matches.

```json
{
  "aggregatable_deduplication_keys": [
    {
      "filters": {"x": ["y"]}
    },
    {
      "deduplication_key": "123",
     "filters": {"x": ["z"]}
    }
  ]
}
```

Reply:
Wouldn't the behavior be the same if the first list entry were omitted in your example?

Reply:
Not exactly. If both filters match, the first matched one will be used. In my example it will be null, but if it's omitted it will be "123".